### PR TITLE
feat: [CDS-47127]: force delete service,env,envGroup

### PR DESCRIPTION
**Pull request by RajBaviskar**
### Summary

- force delete service,env,envGroup

#### Screenshots

Environment
Screen Shot 2023-01-08 at 9 54 35 PM

Service
Screen Shot 2023-01-06 at 3 44 13 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`

- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by RajBaviskar** [comment]
retrigger coverage

**Comment by RajBaviskar** [comment]
retrigger jest

**Comment by RajBaviskar** [comment]
retrigger cypress-pipeline

File: src/services/cd-ng/index.tsx [hunk 1]
```diff
@@ -33732,6 +33732,7 @@ export interface DeleteEnvironmentGroupQueryParams {
   filePath?: string
   commitMsg?: string
   lastObjectId?: string
+  forceDelete?: boolean
```
Comment (AshwinBhatkal-Harness) [inline src/services/cd-ng/index.tsx:33735]: Are we manually doing this?

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentsList/EnvironmentsList.tsx [hunk 2]
```diff
@@ -58,16 +55,37 @@ export default function EnvironmentsList({ response, refetch }: any) {
...
+        setCurEnvId(id)
+        openReferenceErrorDialog()
+      } else {
+        showError(getRBACErrorMessage(e))
+      }
     }
   }
 
+  const redirectToReferencedBy = (): void => {
+    closeDialog()
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentsList/EnvironmentsList.tsx:77]: We have a referenced by section for environments, we can redirect there

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentsList/EnvironmentsList.tsx [hunk 3]
```diff
@@ -34,16 +36,11 @@ export default function EnvironmentsList({ response, refetch }: any) {
   const { showSuccess, showError } = useToaster()
   const { getRBACErrorMessage } = useRBACError()
   const { getString } = useStrings()
+  // const [forceDelete, setforceDelete] = useState(false)
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentsList/EnvironmentsList.tsx:39]: remove commented code

File: src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx [hunk 4]
```diff
@@ -435,35 +437,60 @@ const RenderColumnMenu: Renderer<CellProps<any>> = ({ row, column }) => {
...
+          accountIdentifier: accountId,
+          forceDelete: Boolean(forceDelete)
+        }
+      })
+      if (response.status === 'SUCCESS') {
+        showSuccess(getString('common.deleteServiceMessage'))
+        ;(column as any).reload?.()
+      }
+    } catch (err) {
+      if (err?.data?.code === 'ENTITY_REFERENCE_EXCEPTION') {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx:454]: can we add this same condition for environments/environment groups as well?

File: src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx [hunk 5]
```diff
@@ -435,35 +437,60 @@ const RenderColumnMenu: Renderer<CellProps<any>> = ({ row, column }) => {
...
-          }
-        }
+      if (isConfirmed && data.identifier) {
+        deleteHandler(false)
       }
     }
   })
 
+  const redirectToReferencedBy = (): void => {
+    closeDialog()
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx:482]: We have a referenced by section for services, we can redirect there

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentsList/EnvironmentsList.tsx (hunk 3 above)
Comment (RajBaviskar) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentsList/EnvironmentsList.tsx:39]: Done.

File: src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx (hunk 4 above)
Comment (RajBaviskar) [reply src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx:454]: added.

File: src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx (hunk 5 above)
Comment (RajBaviskar) [reply src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx:482]: I tried to put it in common until but closeDialog def in local on 485. Not sure how to make it completely independent so we reuse it.

**Comment by RajBaviskar** [comment]
retrigger jest

File: src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupsList/__tests__/EnvironmentGroupsList.test.tsx [hunk 6]
```diff
@@ -122,10 +122,6 @@ describe('Environment Groups List View', () => {
       const deleteButton = findDialogContainer()?.querySelectorAll('.bp3-button-text')[0]
       fireEvent.click(deleteButton!)
     })
-
-    await waitFor(() =>
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupsList/__tests__/EnvironmentGroupsList.test.tsx:126]: Can we add the appropriate test for this please?

File: src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx (hunk 5 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/Services/ServicesList/ServicesList.tsx:482]: We don't need close dialog at all. We can just redirect directly

File: src/services/cd-ng/index.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/services/cd-ng/index.tsx]: We should not be doing this manually

Comment (RajBaviskar) [reply src/services/cd-ng/index.tsx]: swagger on UI side? not sure how? want to sync over a zoom call?

File: src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupsList/__tests__/EnvironmentGroupsList.test.tsx (hunk 6 above)
Comment (RajBaviskar) [reply src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupsList/__tests__/EnvironmentGroupsList.test.tsx:126]: updated this test

**Comment by RajBaviskar** [comment]
Not Applicable.

**Pull request closed without merging**